packages/embed/src/utils/hubspot.ts

```diff
@@ -6,23 +8,19 @@ type HubspotFieldsType = {
 
 const HUBSPOT_COOKIE = 'hubspotutk'
 
-/**
- *
- * @returns {string} cookie value
- */
-
 export const getHubspotCookieValue = () => {
   const match = document.cookie.match(new RegExp(`(^| )${HUBSPOT_COOKIE}=([^;]+)`))
   return (match && match[2]) || undefined
 }
 
-/**
- *
- * @returns {HubspotFieldsType} object with all hubspot fields
- */
-
 export const getHubspotHiddenFields = (): HubspotFieldsType => ({
   hubspot_page_name: document.title,
   hubspot_page_url: window.location.href,
   hubspot_utk: getHubspotCookieValue(),
 })
+
+export const waitForHubspotCookie = async (): Promise<void> => {
+  for (let maxRetries = 10, retries = 0; !getHubspotCookieValue() && retries < maxRetries; retries++) {
```

Comment:
It looks like `getHubspotCookieValue()` will get called on every iteration, during the for condition evaluation, is that the idea? It's ok if it is, just thought I'd mention

Reply:
Yes, thats the idea. It is a (hopefully quick) regex check. We get to break out from the "sleep cycle" as soon as the cookie is present.

Reply:
Could this be done with [cookies.onChanged?](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/cookies/onChanged)

Reply:
@mrisoli I see it is currently [not supported in Safari](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/cookies/onChanged#browser_compatibility) 😢